src/responder.ts

```diff
@@ -1,60 +1,271 @@
+            data: data
+        };
+
+        // execute the transaction
+        return this.signer.sendTransaction(transactionRequest);
+    }
+}
+
+
+/* CONCRETE RESPONDER IMPLEMENTATIONS */
 
-            logger.error(
-                `Failed to respond to ${appointment.getEventName()} for appointment ${appointment.getStateLocator()}, after ${tries +
-                    1}. Giving up.`
-            );
-        } catch (bigDoh) {
-            logger.error(bigDoh);
+/**
+ * This responder can only handle one response. The wallet used by this responder should not be used for any other purpose
+ * until the end of the response flow (that is, until the event `responseConfirmed` is emitted).
+ */
+export class EthereumDedicatedResponder extends EthereumResponder {
+    // Waiting time before retrying
+    private static WAIT_TIME_BETWEEN_ATTEMPTS = 1000;
+
+    // Timestamp in milliseconds when the last block was received (or since the creation of this object)
+    private timeLastBlockReceived: number = Date.now();
+    private mAttemptsDone: number = 0;
+
+    get attemptsDone() {
+        return this.mAttemptsDone;
+    }
+
+    /**
+     * @param signer The signer of the wallet associated with this responder. Each responder should have exclusive access to his wallet.
+     * @param appointmentId The id of the Appointment this object is responding to.
+     * @param ethereumResponse The IEthereumResponse containing what needs to be submitted.
+     * @param [confirmationsRequired=40] The number of confirmations required before a transaction is trusted.
+     * @param [maxAttempts=10] The maximum number of retries before the Responder will give up.
+     */
+    constructor(
+        readonly signer: ethers.Signer,
+        public readonly appointmentId: string,
+        public readonly ethereumResponse: IEthereumResponse,
+
+        public readonly confirmationsRequired = 40,
+        private readonly maxAttempts: number = 10
+    ) {
+        super(signer, appointmentId, ethereumResponse);
+
+        signer.provider.on("block", this.updateLastBlockTime);
+    }
+
+    /**
+     * Release any resources used by this instance.
+     */
+    private destroy() {
+        this.signer.provider.removeListener("block", this.updateLastBlockTime);
+    }
+
+    private updateLastBlockTime() {
+        this.timeLastBlockReceived = Date.now();
+    }
+
+    public async respond() {
+        while (this.mAttemptsDone < this.maxAttempts) {
+            this.mAttemptsDone++;
+            try {
+                // Try to call submitStateFunction, but timeout with an error if
+                // there is no response for 30 seconds.
+                const tx = await promiseTimeout(this.submitStateFunction(), 30000);
+
+                // The response has been sent, but should not be considered confirmed yet.
+                this.responseFlow.status = ResponseState.ResponseSent;
+                this.asyncEmit("responseSent", this.responseFlow);
```

Comment:
I should have spotted this before but we should put these magic strings in a string backed enum, like:

```
enum ReponderEvent {
  ResponseSent = "responseSent",
  ResponseConfirmed = "responseConfirmed",
  AttemptFailed = "attemptFailed",
  ResponseFailed = "responseFailed"
}
```